### ISSUE TITLE
Bad icon

### ISSUE THREAD

**Issue by bpabbott** [schoeps/octave_installer#3]
`Octave.svg` did not convert to `octave.ins` properly. Is this normal?

![screen shot 2016-04-26 at 10 14 35 am](https://cloud.githubusercontent.com/assets/231305/14821193/cda215ae-0b97-11e6-9787-0965287ee1d0.png)

**Comment by schoeps** [comment]
The icon looks fine on my machine. The script uses convert (lines 173-182) to create a bunch of png images for the icns. If we want to debug this issue, you could look into the temp folder `tmp_iconset` and check if the png images are ok. 

I am rather sure that convert (from imagemagick) is to blame. I recall that there is a switch "--with-librsvg". I will check if this makes a difference....

![screenshot](https://cloud.githubusercontent.com/assets/868081/14822124/9dc10c06-0bcd-11e6-9c6c-9b94d9cec796.png)

**Comment by bpabbott** [comment]
The icons in `tmp_iconset` are also bad. The switch `--with-librsvg` is present.
![screen shot 2016-04-26 at 11 05 40 am](https://cloud.githubusercontent.com/assets/231305/14823030/faf69a6e-0b9e-11e6-8e03-4dba9a376a5e.png)
I noticed both `graphicsmagick` and `imagemagick` are installed. Is `convert` from `graphicsmagick` also available?

**Comment by bpabbott** [comment]
I don't know why my `convert` isn't working. As a work around, I wrote a simple bash script to generate `octave.icns` using the `pngs` provided by `octave-4.0.1.tag.gz`.

**Comment by schoeps** [comment]
Can you post it here? It's probably very simple indeed but it may save me some time. 

I think there was a good reason why I recreated those images but I do not recall why :)

**Comment by bpabbott** [comment]
Below is a snippet of the code I used.

> hicolor="$install_dir/Contents/Resources/usr/opt/octave/share/icons/hicolor"
> svg_icon="$hicolor/scalable/apps/octave.svg"
> tmp_iconset="$(mktemp -d /tmp/iconset-XXXX)/droplet.iconset"
> mkdir -p "$tmp_iconset"
> cp "$hicolor/16x16/apps/octave.png" "$tmp_iconset/icon_16x16.png"
> cp "$hicolor/32x32/apps/octave.png" "$tmp_iconset/icon_16x16@2x.png"
> cp "$hicolor/32x32/apps/octave.png" "$tmp_iconset/icon_32x32.png"
> cp "$hicolor/64x64/apps/octave.png" "$tmp_iconset/icon_32x32@2x.png"
> cp "$hicolor/128x128/apps/octave.png" "$tmp_iconset/icon_128x128.png"
> cp "$hicolor/256x256/apps/octave.png" "$tmp_iconset/icon_128x128@2x.png"
> cp "$hicolor/256x256/apps/octave.png" "$tmp_iconset/icon_256x256.png"
> cp "$hicolor/512x512/apps/octave.png" "$tmp_iconset/icon_256x256@2x.png"
> cp "$hicolor/512x512/apps/octave.png" "$tmp_iconset/icon_512x512.png"
> iconutil -c icns -o "$install_dir/Contents/Resources/octave.icns" "$tmp_iconset"

**Comment by bpabbott** [comment]
Are the 2x.png versions needed? If not ...

> hicolor="$install_dir/Contents/Resources/usr/opt/octave/share/icons/hicolor"
> svg_icon="$hicolor/scalable/apps/octave.svg"
> tmp_iconset="$(mktemp -d /tmp/iconset-XXXX)/droplet.iconset"
> mkdir -p "$tmp_iconset"
> cp "$hicolor/16x16/apps/octave.png" "$tmp_iconset/icon_16x16.png"
> cp "$hicolor/32x32/apps/octave.png" "$tmp_iconset/icon_32x32.png"
> cp "$hicolor/128x128/apps/octave.png" "$tmp_iconset/icon_128x128.png"
> cp "$hicolor/256x256/apps/octave.png" "$tmp_iconset/icon_256x256.png"
> cp "$hicolor/512x512/apps/octave.png" "$tmp_iconset/icon_512x512.png"
> iconutil -c icns -o "$install_dir/Contents/Resources/octave.icns" "$tmp_iconset"

**Comment by bpabbott** [comment]
Regarding `librsvg`, I don't see that library in the `lib` directory. Does `librsvg` need to be installed independently for `imagemagick` to use it? Do you see it in `$install_dir/Contents/Resources/usr/lib`?

I'm installing `librsvg` now, and will try reinstalling `imagemagick` after it is installed, and then try to generate the icons with `convert`.

**Comment by schoeps** [comment]
Don't worry. I think I will add your method (as far as I recall the pngs are only problematic if you compile the development version).

**Comment by schoeps** [comment]
Fixed in e4513c2e0ac2f8e3b3867a2f7fe05e105c87141d

**Comment by bpabbott** [comment]
umph ... I'm seeing the effort below.

```
Copying volume icon file '/Applications/Octave.app/Contents/Resources/octave.icns'...
cp: /Applications/Octave.app/Contents/Resources/octave.icns: No such file or directory
```

**Comment by schoeps** [comment]
I forget to rename "octave.icns" to "applet.icns" when creating the dmg. Fixed in 73de383d8a41dd16df981d9b755a97cb6fd79069
